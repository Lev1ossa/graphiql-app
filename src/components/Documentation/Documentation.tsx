import { getApiIntrospectionSchema } from '@/API'
import { useEnqueueSnackbar } from '@/hooks/useEnqueueSnackbar'
import { useBoundStore } from '@/store'
import { Stack, TextField } from '@mui/material'
import { useEffect, useState } from 'react'

export default function Documentation() {
  const baseUrl = useBoundStore((state) => state.baseUrl)
  const [documentation, setDocumentation] = useState('')

  const { pushSnackbar } = useEnqueueSnackbar()

  useEffect(() => {
    setDocumentation('')
    getApiIntrospectionSchema(baseUrl).then(
      (result) => setDocumentation(result),
      (error: Error) => pushSnackbar({ message: error.message }),
    )
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [baseUrl])
  return (
    <Stack direction="row">
      <TextField sx={{ width: '100%' }} multiline value={documentation} />
    </Stack>
  )
}
