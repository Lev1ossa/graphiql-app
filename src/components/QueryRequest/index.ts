export { default as QueryRequest } from './QueryRequest'
