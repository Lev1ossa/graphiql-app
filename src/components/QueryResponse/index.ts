export { default as QueryResponse } from './QueryResponse'
