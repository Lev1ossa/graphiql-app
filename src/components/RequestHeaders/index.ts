export { default as RequestHeaders } from './RequestHeaders'
