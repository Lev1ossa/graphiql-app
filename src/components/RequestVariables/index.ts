export { default as RequestVariables } from './RequestVariables'
