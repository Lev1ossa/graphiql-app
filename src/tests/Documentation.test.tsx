import { Documentation } from '@/components/Documentation'
import { expect, test } from 'vitest'
import renderInContext from './renderInContext'

test('Correct Documentation snapshot', () => {
  const { baseElement } = renderInContext(<Documentation />)
  expect(baseElement).toMatchSnapshot()
})
