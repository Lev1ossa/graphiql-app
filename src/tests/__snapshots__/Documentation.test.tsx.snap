// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct Documentation snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiStack-root css-m69qwo-MuiStack-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</body>
`;
